src/dataflow-types/src/types.rs

```diff
@@ -622,6 +622,31 @@ pub enum Compression {
     None,
 }
 
+/// The meaning of the timestamp number produced by data sources.
+#[derive(Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Serialize, Deserialize, Hash)]
+pub enum Timeline {
+    /// The system timeline is used by the mz_ catalog items that perform internal
+    /// bookkeeping. Although this generally corresponds to wall-clock-now, we
+    /// have chosen for now to make this a separate timeline. If users complain and
+    /// there's a valid use case to join these with RealTime sources, we can change
+    /// this in the future.
+    System,
+    /// RealTime means the timestamp is the number of milliseconds since the Unix
+    /// epoch.
+    RealTime,
+    /// BringYourOwn is for Debezium consistency topics.
+    BringYourOwn(BringYourOwn),
+}
+
+/// A struct to hold more specific information about where a BYO source
+/// came from so we can differentiate between topics of the same name across
+/// different brokers.
+#[derive(Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Serialize, Deserialize, Hash)]
+pub struct BringYourOwn {
+    pub broker: String,
```

Comment:
does this work right if the topic is replicated across multiple brokers? (I think it should be fine because we specify a single broker host in the create source and thats what all instances will check with)

Reply:
also, a more serious question -- I think if I understand the code right this means that every single Debezium source is on its own timeline and can't be joined with anything else -- is that right?

Reply:
I don't know anything about kafka brokers, but this is what Chris told me I'd need to do to make it unique? It's possible there's other problems. I tried to make this a separate commit so someone who knows what this is could easily figure out if it's correct.

You are correct, separate debezium topics should not be joinable.

Reply:
yeah, I think we are going to want something a little bit coarser than (broker, topic) because then at that point i think it means "you can only do self-joins with this source" iiuc.

the idea with debezium is that it takes a database and replicates a set of tables st each table gets copied onto a Kafka topic. I think we would want to have a timeline tied to the upstream database name or id or something like that, so that we can express "these two sources are different, but Debezium has assigned them both meaningful similar timestamps because they come from the same upstream database"

Reply:
yup yup